spec/controllers/works/default_rails_actions_spec.rb

```diff
@@ -277,30 +327,55 @@ def call_with_params(params)
   end
 
   describe "index" do
-    before do
-      @fandom = create(:canonical_fandom)
-      @work = create(:work, fandom_string: @fandom.name)
-    end
+    let(:fandom) { create(:canonical_fandom) }
+    let!(:work) { create(:work, fandom_string: fandom.name) }
 
     it "returns the work" do
       get :index
-      expect(assigns(:works)).to include(@work)
+      expect(assigns(:works)).to include(work)
     end
 
     it "sets the fandom when given a fandom id" do
-      params = { fandom_id: @fandom.id }
+      params = { fandom_id: fandom.id }
       get :index, params: params
-      expect(assigns(:fandom)).to eq(@fandom)
+      expect(assigns(:fandom)).to eq(fandom)
+    end
+
+    it "redirects to tag page for noncanonical tags" do
+      unsorted_tag = create(:unsorted_tag)
```

Comment:
Could we use a different kind of tag here? When you use a tag on a work in reality, it will always be a rating, category, fandom, warning, freeform, character, or relationship. Unsorted tags only exist on bookmarks -- they're tags that haven't already been classified as fandom, freeform, etc.

Reply:
Fixed